src/util-base64.c

```diff
@@ -103,7 +103,6 @@ uint32_t DecodeBase64(uint8_t *dest, const uint8_t *src, uint32_t len) {
             /* Invalid character found, so decoding fails */
             if (src[i] != '=') {
                 valid = 0;
-                numDecoded = 0;
```

Comment:
Yeah, it might have an effect as the mime decoder is using this to detect an anomaly in the base64 data, whereas base64_decode sometimes will just want to decode as much base64 as it can until a bad byte.

Toughts on adding a new flag, strict to determine if it should fail, or return what its decoded so far?